src/App.js

```diff
@@ -158,7 +167,11 @@ function App() {
                 isOpen={state.isModalOpen}
                 setIsOpen={(isOpen) => dispatch({ type: 'setModalState', payload: isOpen })}
                 coordinates={state.userCoordinates}
-                setCoordinates={(coords) => dispatch({ type: 'setUserCoordinates', payload: coords })}
+                setCoordinates={(coords) => {
+                  // Save the user coordinates in order to reuse them on the next user session
+                  setUserCoordinatesToLocalStorage(coords);
```

Comment:
I'd put this in the handler for the setUserCoordinates event in the reducer. That way it will be called even if the event is dispatched from someplace else

Reply:
I thought about that as well, and putting it in the reducer creates a side-effect in the reducer which is generally something I prefer to avoid. You can create an action that both does the side-effect and fire the event. Specifically in this case, it creates a situation where we set the coordinates to local storage just after we retrieved them from local storage, which is a noop but still doesn't make a lot of sense.